### DISCUSSION TITLE
Not finding a reliable prompt to call MemoryPlugin

### DISCUSSION THREAD

**Discussion by henrystaples** [microsoft/semantic-kernel#5833] [Q&A] [answered]
Are there any settings in the Kernel that can affect how a MemoryPlugin would answer? 
Or, rather, should I look for a prompt that could reliably call the Kernel Memory Plugin and receive an answer.

If I call the a MemoryPlugin.AskAsync(prompt), I get back a reasonable answer.
But, if I call kernel.InvokePromptAsync(prompt), with the same prompt, sometimes I get back "I don't know."

I followed some examples and setup a MemoryPlugin.
`var plugin = new MemoryPlugin(kernelMemory, waitForIngestionToComplete: true);`
`kernel.ImportPluginFromObject(plugin, "memory");`

In this example, both memoryResponse and response work:

### For this prompt & message, both AskAsync and InvokePromptAsync work
             var message = @"What did the fox jump over?";
             var prompt = $@"{{memory.ask}} {message} 
                          If Kernel Memory doesn't know the answer, say 'I don't know'.

                          Question to Kernel Memory:  { message }";
             var memoryResponse = await plugin.AskAsync(prompt);
             var response = await kernel.InvokePromptAsync(prompt);

**********************
**prompt:**
{memory.ask} What did the fox jump over? 
  If Kernel Memory doesn't know the answer, say 'I don't know'.

  Question to Kernel Memory:  What did the fox jump over?
**memoryResponse:**
The quick brown fox jumped over the lazy dog.
**response:**
The fox jumped over the lazy dog.

### But when I change the message, it's hit-or-miss with the kernel.invokePromptAsync method.
             var message = @"Describe the dissimilarity index for Horry County, SC";
**prompt:**
{memory.ask} Describe the dissimilarity index for Horry County, SC 
If Kernel Memory doesn't know the answer, say 'I don't know'.

Question to Kernel Memory:  Describe the dissimilarity index for Horry County, SC
**memoryResponse:**
HUD Table 12 provides Opportunity Indicators by Race/Ethnicity for Horry County, SC. The table includes the School Proficiency Index, which measures the proficiency of elementary schools in the jurisdiction and is determined by the performance of 4th graders on state exams. The scores in the jurisdiction are relatively similar among the different racial and ethnic groups in the area, with one exception. The Black, Non-Hispanic population has an index score of 44.36, while all other groups have scores between 62.54 and 69.49. This points to disparity in access to educational opportunities for the Black, Non-Hispanic population in the jurisdiction.
**response:**
I don't know.

Any ideas what I could tweak to get a better hit from memory when using kernel.InvokePromptAsync()?

**Comment by henrystaples** [comment, marked as answer]
FWIW, I moved to using a Handlebars Planner and am finding results reliably now.

**Comment by aherrick** [comment]
@henrystaples could you share a snippet of your code? i'm looking for something similar to your issue and not sure what you mean by handlebar planner?
